app/views/Settings/LocationTrackingStatus.js

```diff
@@ -0,0 +1,184 @@
+import styled, { css } from '@emotion/native';
+import React, { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Linking } from 'react-native';
+import DeviceSettings from 'react-native-device-settings';
+import { openSettings } from 'react-native-permissions';
+
+import { Button, Divider, Switch, Typography } from '../../components';
+import Colors from '../../constants/colors';
+import { config } from '../../COVIDSafePathsConfig';
+import { useLocTrackingStatus } from '../../services/hooks/useLocTrackingStatus';
+import { Reason } from '../../services/LocationService';
+import { isPlatformiOS } from '../../Util';
+
+export const LocationTrackingStatus = () => {
+  const { i18n } = useTranslation();
+  const [locTrackingStatus, setLocTrackingStatus] = useLocTrackingStatus();
+  console.log(locTrackingStatus);
+
+  const { reason, canTrack } = locTrackingStatus;
+
+  const isGPS = config.tracingStrategy === 'gps';
+
+  const hasLocTrackingPermissions =
+    reason === Reason.USER_ENABLED || reason === Reason.USER_DISABLED;
+
+  const isMissingLocPermissions =
+    reason === Reason.DEVICE_LOCATION_OFF ||
+    reason === Reason.APP_NOT_AUTHORIZED;
+
+  const getCurrentRowDirection = i18n.dir() === 'rtl' ? 'row-reverse' : 'row';
+
+  const getLocStatusElement = () => {
+    console.log(`reason: ${reason}`);
+    console.log(
+      `hasLocTrackingPermissions: ${hasLocTrackingPermissions} isMissingLocPermissions: ${isMissingLocPermissions}`,
+    );
+
+    if (hasLocTrackingPermissions) {
```

Comment:
Already inside the locStatus component, this could be just hasPermissions.

Personally, prefer zero abbreviations outside of well known acronyms. The code is minified, so you'll save the space in the real app.

Reply:
Yup - `loc` feels awkward all around.